src/h2load.cc

```diff
@@ -684,62 +821,43 @@ void Client::on_header(int32_t stream_id, const uint8_t *name, size_t namelen,
   }
 }
 
-void Client::on_status_code(int32_t stream_id, uint16_t status) {
-  auto itr = streams.find(stream_id);
-  if (itr == std::end(streams)) {
-    return;
-  }
-  auto &stream = (*itr).second;
-
-  if (status >= 200 && status < 300) {
-    ++worker->stats.status[2];
-    stream.status_success = 1;
-  } else if (status < 400) {
-    ++worker->stats.status[3];
-    stream.status_success = 1;
-  } else if (status < 600) {
-    ++worker->stats.status[status / 100];
-    stream.status_success = 0;
-  } else {
-    stream.status_success = 0;
-  }
-}
-
 void Client::on_stream_close(int32_t stream_id, bool success, bool final) {
-  ++req_done;
-  --req_inflight;
+  if (warmup == 2) {
+    ++req_done;
+    if (req_inflight > 0) {
```

Comment:
This got changed because we reset the `req_inflight` to `0` after the warmup. Also, there may be other instances where it may come to `0`. Therefore it needs to decrease only when it's more than `0`. So, I added this safety check.